webapp/e2e/cypress/support/index.ts

```diff
@@ -1,9 +1,219 @@
+import 'mattermost-webapp/e2e/cypress/support/ui';
+import 'mattermost-webapp/e2e/cypress/support/ui_commands'; // soon to deprecate
+import 'mattermost-webapp/e2e/cypress/support/visual_commands';
+import 'mattermost-webapp/e2e/cypress/support/external_commands';
+
+Cypress.on('test:after:run', (test, runnable) => {
+    // Only if the test is failed do we want to add
+    // the additional context of the screenshot.
+    if (test.state === 'failed') {
+        let parentNames = '';
+
+        // Define our starting parent
+        let parent = runnable.parent;
+
+        // If the test failed due to a hook, we have to handle
+        // getting our starting parent to form the correct filename.
+        if (test.failedFromHookId) {
+            // Failed from hook Id is always something like 'h2'
+            // We just need the trailing number to match with parent id
+            const hookId = test.failedFromHookId.split('')[1];
+
+            // If the current parentId does not match our hook id
+            // start digging upwards until we get the parent that
+            // has the same hook id, or until we get to a tile of ''
+            // (which means we are at the top level)
+            if (parent.id !== `r${hookId}`) {
+                while (parent.parent && parent.parent.id !== `r${hookId}`) {
+                    if (parent.title === '') {
+                        // If we have a title of '' we have reached the top parent
+                        break;
+                    } else {
+                        parent = parent.parent;
+                    }
+                }
+            }
+        }
+
+        // Now we can go from parent to parent to generate the screenshot filename
+        while (parent) {
+            // Only append parents that have actual content for their titles
+            if (parent.title !== '') {
+                parentNames = parent.title + ' -- ' + parentNames;
+            }
+
+            parent = parent.parent;
+        }
+
+        // Clean up strings of characters that Cypress strips out
+        const charactersToStrip = /[;:"<>/]/g;
+        parentNames = parentNames.replace(charactersToStrip, '');
+        const testTitle = test.title.replace(charactersToStrip, '');
+
+        // If the test has a hook name, that means it failed due to a hook
+        // and consequently Cypress appends some text to the file name
+        const hookName = test.hookName ? ' -- ' + test.hookName + ' hook' : '';
+
+        const filename = encodeURIComponent(`${parentNames}${testTitle}${hookName} (failed).png`);
+
+        // Add context to the mochawesome report which includes the screenshot
+        addContext({test}, {
+            title: 'Failing Screenshot: >> screenshots/' + Cypress.spec.name + '/' + filename,
+            value: 'screenshots/' + Cypress.spec.name + '/' + filename,
+        });
+    }
+});
+
+before(() => {
+    // # Try to login using existing sysadmin account
```

Comment:
Is this functionality that can be imported from the webapp?

Reply:
Yes, but the babel/webpack issue needs to be solved so we can just import `import 'mattermost-webapp/e2e/cypress/support` without any other code needed in this file.

https://github.com/mattermost/mattermost-plugin-demo/pull/136#discussion_r544425708

Reply:
I see